File: 66.plus-one.java
/*
 * @lc app=leetcode id=66 lang=java
 *
 * [66] Plus One
 */

// @lc code=start
class Solution {
    public int[] plusOne(int[] digits) {
        if (digits == null || digits.length == 0) { // Kiểm tra mảng rỗng hoặc null
            return new int[]{1}; // Trả về mảng chứa 1 nếu mảng rỗng
        }
        int n = digits.length;
        for (int i = n - 1; i >= 0; i--) { // Duyệt từ phải sang trái
            if (digits[i] < 9) { // Nếu phần tử nhỏ hơn 9
                digits[i]++; // Tăng phần tử lên 1
                return digits; // Trả về mảng đã cập nhật
            }
            digits[i] = 0; // Nếu phần tử bằng 9, đặt thành 0
        }
        // Nếu tất cả các phần tử đều là 9
        int[] newNumber = new int[n + 1];
        newNumber[0] = 1; // Đặt phần tử đầu tiên là 1
        return newNumber; // Trả về mảng mới
    }
}
// @lc code=end

File: 66.plus-one.py
#
# @lc app=leetcode id=66 lang=python3
#
# [66] Plus One
#

# @lc code=start
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
            n = len(digits) # Lấy độ dài của danh sách
            for i in range(n - 1, -1, -1): # Duyệt từ phải sang trái
                if digits[i] < 9: # Nếu phần tử nhỏ hơn 9
                    digits[i] += 1 # Tăng giá trị phần tử lên 1
                    return digits # Trả về danh sách đã cập nhật
                digits[i] = 0 # Đặt giá trị phần tử thành 0
            return [1] + digits # Thêm 1 vào đầu danh sáchd
# @lc code=end